workspaces/copilot/plugins/copilot-common/src/index.ts

```diff
@@ -176,3 +178,173 @@ export interface TeamInfo {
    */
   name: string;
 }
+
+/**
+ * Thin wrapper for the CopilotMetrics type
+ * @public
+ */
+export interface CopilotLanguages {
+  /**
+   * The language name
+   */
+  name: string;
+  /**
+   * The number of engaged users
+   */
+  total_engaged_users: number;
+  /**
+   * total number of code suggestions
+   */
+  total_code_suggestions: number;
+  /**
+   * total number of code acceptances
+   */
+  total_code_acceptances: number;
+  /**
+   * total number of code lines suggested
+   */
+  total_code_lines_suggested: number;
+  /**
+   * total number of code lines accepted
+   */
+  total_code_lines_accepted: number;
+}
+
+/**
+ * Thin wrapper for the CopilotMetrics type
+ * @public
+ */
+export interface CopilotModels {
+  /**
+   * The model name
+   */
+  name: string;
+  /**
+   * List of languages this model was used in
+   */
+  languages: CopilotLanguages[];
+}
+
+/**
+ * Thin wrapper for the CopilotMetrics type
+ * @public
+ */
+export interface CopilotEditors {
+  /**
+   * The editor name
+   */
+  name: string;
+  /**
+   * List of models this editor was used in
+   */
+  models: CopilotModels[];
+}
+
+/**
+ * Thin wrapper for the CopilotMetrics type
+ * @public
+ */
+export interface CopilotIdeCodeCompletions {
+  editors: CopilotEditors[];
+}
+
+/**
+ * Represents the metrics data for copilot
+ *
+ * Not entirely implemented. Only the stuff we needed to satisfy #2828
+ *
+ * https://docs.github.com/en/rest/copilot/copilot-metrics?apiVersion=2022-11-28
+ *
+ * @public
+ */
+export interface CopilotMetrics {
+  /**
+   * The date for the metrics reported.
+   */
+  date: string;
+
+  /**
+   * The total number of active users.
+   */
+  total_active_users: number;
+
+  /**
+   * The total number of engaged users.
+   */
+  total_engaged_users: number;
+
+  /**
+   * The total number of code suggestions for IDE users.
+   */
+  copilot_ide_code_completions: CopilotIdeCodeCompletions;
+}
+
+/**
+ * @internal
+ * @deprecated This is just a bridge function to make #2828 work on short notice, long term, we should rely on CopilotMetrics instead of Metric
+ **/
+export function convertToMetric(
```

Comment:
If this is going away then why not have it in the `GithubClient` and call it in the functions that need it so they just return this for now. Would probably simplify things and then you would not need to export this.

Reply:
I don't think `fetchEnterpriseCopilotMetrics: () => Promise<CopilotMetrics[]>;` should return a `Metrics[]`, the new structure should be used and encouraged for new things. It's just existing things that benefit from this conversion without refactoring everything.

I did like the idea of moving it from -common to the -backend, as it's not really used anywhere else.